testing/trino-product-tests/src/main/java/io/trino/tests/product/iceberg/TestIcebergSparkCompatibility.java

```diff
@@ -2528,10 +2529,11 @@ public void testRegisterTableWithTableLocation(StorageFormat storageFormat)
 
         List<Row> expected = List.of(row(1, "INDIA", true), row(2, "USA", false));
         String tableLocation = getTableLocation(trinoTableName);
+        String metadataLocation = getLatestMetadataLocation(TRINO_CATALOG, TEST_SCHEMA_NAME, baseTableName);
         // Drop table from hive metastore and use the same table name to register again with the metadata
         dropTableFromMetastore(baseTableName);
 
-        onTrino().executeQuery(format("CALL iceberg.system.register_table ('%s', '%s', '%s')", TEST_SCHEMA_NAME, baseTableName, tableLocation));
+        onTrino().executeQuery(format("CALL iceberg.system.register_table ('%s', '%s', '%s', '%s')", TEST_SCHEMA_NAME, baseTableName, tableLocation, metadataLocation));
```

Comment:
The test name is `testRegisterTableWithTableLocation` and we have `testRegisterTableWithMetadataFile`. Specifying metadata file name in this test looks weird to me.